File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    let n = new Set(nums);
    let maxLen = 0;

    for(let num of n){
        if(!n.has(num-1)){
            let curr = num;
            let len = 1;
            while(n.has(curr+1)){
                curr += 1;
                len += 1;
            }
            maxLen = Math.max(maxLen, len);
        }
    }

    return maxLen;
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.go
func longestConsecutive(nums []int) int {
    set := make(map[int]bool)
    for _, num := range nums {
        set[num] = true
    }

    max := 0

    for num := range set {
        if !set[num-1] {
            curr := num
            len := 1

            for set[curr+1] {
                curr++
                len++
            }

            if len > max {
                max = len
            }
        }
    }

    return max
}